healthcare/api-client/v1/fhir/fhir_resources_test.py

```diff
@@ -0,0 +1,282 @@
+        except HttpError as err:
+            # We ignore 409 conflict here, because we know it's most
+            # likely the first request failed on the client side, but
+            # the creation suceeded on the server side.
+            if err.resp.status == 409:
+                print(
+                    "Got exception {} while creating FHIR store".format(err.resp.status)
+                )
+            else:
+                raise
+
+    create()
+
+    yield
+
+    # Clean up
+    @backoff.on_exception(backoff.expo, HttpError, max_time=60)
+    def clean_up():
+        try:
+            fhir_stores.delete_fhir_store(
+                project_id, cloud_region, dataset_id, fhir_store_id
+            )
+        except HttpError as err:
+            # The API returns 403 when the FHIR store doesn't exist.
+            if err.resp.status == 404 or err.resp.status == 403:
+                print(
+                    "Got exception {} while deleting FHIR store".format(err.resp.status)
+                )
+            else:
+                raise
+
+    clean_up()
+
+
+# Fixture that creates/deletes a Patient resource for various tests.
+@pytest.fixture(scope="module")
+def test_patient():
+    patient_response = fhir_resources.create_patient(
+        base_url, project_id, cloud_region, dataset_id, fhir_store_id,
+    )
+    patient_resource = patient_response.json()
+    patient_resource_id = patient_resource["id"]
+
+    yield patient_resource_id
+
+    @backoff.on_exception(backoff.expo, HttpError, max_time=60)
+    # Clean up
+    def clean_up():
+        try:
+            fhir_resources.delete_resource(
+                base_url,
+                project_id,
+                cloud_region,
+                dataset_id,
+                fhir_store_id,
+                resource_type,
+                patient_resource_id,
+            )
+
+        except HttpError as err:
+            # The API returns 200 whether the resource exists or was
+            # successfully deleted or not, so only retry on
+            # unathorized exceptions.
+            if err.resp.status > 200:
+                print(
+                    "Got exception {} while deleting FHIR store".format(err.resp.status)
+                )
+            else:
+                raise
+
+    clean_up()
+
+
+def test_create_patient(test_dataset, test_fhir_store, capsys):
+    # Manually create a new Patient here to test that creating a Patient
+    # works.
+    fhir_resources.create_patient(
+        base_url, project_id, cloud_region, dataset_id, fhir_store_id,
+    )
+
+    out, _ = capsys.readouterr()
+
+    assert "Created Patient" in out
+
+
+def test_get_patient(test_dataset, test_fhir_store, test_patient, capsys):
+    fhir_resources.get_resource(
+        base_url,
+        project_id,
+        cloud_region,
+        dataset_id,
+        fhir_store_id,
+        resource_type,
+        test_patient,
+    )
+
+    out, _ = capsys.readouterr()
+
+    print(out)
```

Comment:
nit: you don't need this I guess.

Reply:
Weird, I don't know where these statements came from.

Reply:
I still see `print(out)`